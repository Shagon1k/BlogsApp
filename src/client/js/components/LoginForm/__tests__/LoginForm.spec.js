import React from 'react';
import LoginPage from '../index.jsx';

describe('LoginForm component', () => {
	const onLogin = jest.fn();

	it('renders "You are logged in!" if user is logged in', () => {
		const wrapper = shallow(<LoginPage isLoggedIn={true} onLogin={onLogin}/>);
		expect(wrapper).toMatchSnapshot();

	});
	
	it('renders login form if user is not logged in', () => {
		const wrapper = shallow(<LoginPage isLoggedIn={false} onLogin={onLogin}/>);
		expect(wrapper).toMatchSnapshot();

		wrapper.find('.loginBtn').simulate('click');
		expect(onLogin).toBeCalled();
	})
})