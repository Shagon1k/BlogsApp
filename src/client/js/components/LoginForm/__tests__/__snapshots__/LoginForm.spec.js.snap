// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders "You are logged in!" if user is logged in 1`] = `
<div
  className="loginFormContainer"
>
  <span
    className="infoText"
  >
    You are logged in!
  </span>
  <Link
    className="lnk"
    replace={false}
    to="/"
  >
    Home Page
  </Link>
</div>
`;

exports[`LoginForm component renders login form if user is not logged in 1`] = `
<div
  className="loginFormContainer"
>
  <form
    className="loginForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="inputUsername"
    >
      Username:
    </label>
    <input
      id="inputUsername"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="inputPassword"
    >
      Password:
    </label>
    <input
      id="inputPassword"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="loginBtn btn"
    >
      Login
    </button>
  </form>
  <Link
    className="lnk"
    replace={false}
    to="/"
  >
    Home Page
  </Link>
</div>
`;
